### ISSUE TITLE
Display error Date when set labels for xaxis

### ISSUE THREAD

**Issue by cuobiezi** [apexcharts/apexcharts.js#4763]
### Steps to Reproduce

1.  Config chart type as 'line'. and xaxis type is 'datetime'
2.  Config max as 1714881301463, min as 1714838101463, tickAmount as 10.
3.  Display xaixs time start  close to next day (eg: Sat May 04 2024 23:55:01).
4.  Config labels as 
```js
labels: {
      datetimeUTC: false,
      formatter: undefined,
      datetimeFormatter: {
        year: 'yyyy',
        month: 'MMM yyyy',
        day: 'dd MMM',
        hour: 'HH:mm'
      }
    }
```

### Reproduce Env

- OS version: Edition Windows 10 Enterprise LTSC
- Chrome version: 129.0.6668.70 (Official Build) (64-bit) 
- Chrome version:  87.0.4280.141 (Official Build) (64-bit)

I can reproduce both of above chrome version .

### Expected Behavior
Display correct Date **4 May** at the left of **xaxis label** .

### Actual Behavior
Display incorrect Date **24 May** the the left of **xaxis label**.

### Screenshots
![image](https://github.com/user-attachments/assets/ee8e2ac0-e1bc-4deb-a549-43b98b7bd097)
![image](https://github.com/user-attachments/assets/76c5f4c2-a800-4d99-8efc-0f4d4e2b61be)

### Reproduction Link
https://codepen.io/cuobiezi/pen/MWNpqJr

**Comment by brianlagunas** [comment]
your reproduction link does not reproduce the issue. Nothing is being rendered on the chart as it seems the data is not correct. Please verify the reproduction code.

**Comment by cuobiezi** [comment]
> your reproduction link does not reproduce the issue. Nothing is being rendered on the chart as it seems the data is not correct. Please verify the reproduction code.

Updated data. 
Actually, My issue didn't relate to data , It is xaxis label display incorrect.  Config Max Time 1714881301463 is **5 May**  ,Min time 1714838101463 is **4 May**. But the xaxis label display start form **24 May**, not in the correct range **4 May - 5 May**

**Comment by cuobiezi** [comment]
> I am not seeing the issue you are describing. Can you provide a screenshot of what you are seeing?

Updated in the description, Thanks.

**Comment by junedchhipa** [comment]
@cuobiezi What browser are you using? I am unable to see what's in the screenshot.

**Comment by cuobiezi** [comment]
> @cuobiezi What browser are you using? I am unable to see what's in the screenshot.
@junedchhipa Updated reproduce env in the Description , Thanks.

**Comment by cuobiezi** [comment]
> @cuobiezi What browser are you using? I am unable to see what's in the screenshot.

@junedchhipa Updated reproduce env in the Description , Thanks.

**Comment by brianlagunas** [comment]
This is what the code pen shows me on Windows and Chrome version 129.0.6668.101 

![image](https://github.com/user-attachments/assets/1d9e815a-7b2c-4dc2-9643-9257305d5d7d)

**Comment by cuobiezi** [comment]
> This is what the code pen shows me on Windows and Chrome version 129.0.6668.101
> 
> ![image](https://private-user-images.githubusercontent.com/835562/377916029-1d9e815a-7b2c-4dc2-9643-9257305d5d7d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NjU4MTgsIm5iZiI6MTcyOTU2NTUxOCwicGF0aCI6Ii84MzU1NjIvMzc3OTE2MDI5LTFkOWU4MTVhLTdiMmMtNGRjMi05NjQzLTkyNTczMDVkNWQ3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMlQwMjUxNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDYxZjE1ZGE3MTJjNDMzYjQ1ZTQ0MWMwZGMwYjUxN2QzNDgwYzE4ZDE1MjQ5YjUxOGMzZTk3MDk5NDIzMmMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.A_p56G9wiwyUilGr9yK0SqWX2oxw_TE2YqDue2Cxs-s)

Your screen xaxis start from **4 May** morning.  The reproduce case xaixs min config field **1714838101463** the xaixs should be from ** 4 May ** start evening. 
![image](https://github.com/user-attachments/assets/e8f42227-be3d-4d90-8e67-1680b88f3124)

**Comment by cuobiezi** [comment]
> This is what the code pen shows me on Windows and Chrome version 129.0.6668.101
> 
> ![image](https://private-user-images.githubusercontent.com/835562/377916029-1d9e815a-7b2c-4dc2-9643-9257305d5d7d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NjcwNzksIm5iZiI6MTcyOTU2Njc3OSwicGF0aCI6Ii84MzU1NjIvMzc3OTE2MDI5LTFkOWU4MTVhLTdiMmMtNGRjMi05NjQzLTkyNTczMDVkNWQ3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMlQwMzEyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMDlhNWFkZGU3ZjIyYWFiMTIyMTBmM2M4NTRhYzgzZjI3NzYxNWY3MTI5NmU4NGVhY2NlN2YwNDdiMzk5Yjg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RHjvChagkGkkW8hCpem3QRQakXOIiXvqmQaX6K_J1XU)

Oh , I got it , we are difference time zone, LOL. The reproduce key step is set the xaixs start time close to next day .

**Comment by cuobiezi** [comment]
> @cuobiezi What browser are you using? I am unable to see what's in the screenshot.

that because time zone issue, Please refer step 3 in description

**Comment by cuobiezi** [comment]
> > This is what the code pen shows me on Windows and Chrome version 129.0.6668.101
> > ![image](https://private-user-images.githubusercontent.com/835562/377916029-1d9e815a-7b2c-4dc2-9643-9257305d5d7d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1NjcwNzksIm5iZiI6MTcyOTU2Njc3OSwicGF0aCI6Ii84MzU1NjIvMzc3OTE2MDI5LTFkOWU4MTVhLTdiMmMtNGRjMi05NjQzLTkyNTczMDVkNWQ3ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyMlQwMzEyNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMDlhNWFkZGU3ZjIyYWFiMTIyMTBmM2M4NTRhYzgzZjI3NzYxNWY3MTI5NmU4NGVhY2NlN2YwNDdiMzk5Yjg0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RHjvChagkGkkW8hCpem3QRQakXOIiXvqmQaX6K_J1XU)
> 
> Oh , I got it , we are difference time zone, LOL. The reproduce key step is set the xaixs start time close to next day . like (Sat May 04 2024 23:55:01)

**Comment by Leon0824** [comment]
I have a similiar issue.

![localhost_5173_system_stations_58CC6D5ED773131B (1)](https://github.com/user-attachments/assets/e4628730-34b8-47fa-8e76-a2d6859fc1e2)

The label '12-31 Tue` should be '11-30 Sun'. You may reproduce this with this dataset:

```json
[
    { "x": "2024-11-30T23:50:00.000Z", "y": 0 },
    { "x": "2024-12-01T00:00:00.000Z", "y": 10 },
    { "x": "2024-12-02T00:00:00.000Z", "y": 20 },
    { "x": "2024-12-02T00:30:00.000Z", "y": 0 }
]
```

If the first entry is more far away from `12-01 00:00:00`, the date label is correct.

**Comment by cuobiezi** [comment]
@Leon0824  fixed in 4.1.0. 
@junedchhipa  @brianlagunas @franklx  But when I upgrade to 4.1.0 will get an error https://github.com/apexcharts/apexcharts.js/discussions/4848, Would you help to check ?
